### PR TITLE
Handle duplicate EXIF header

### LINKED ISSUE #8348: Error reading EXIF from file: SyntaxError: not a TIFF file (header b'' not valid)
### What did you do?

Tried to load image (attached below) and rotate it using `image = ImageOps.exif_transpose(image)`
Related to #4852  and #6123 

### What did you expect to happen?
Image rotated according to EXIF

### What actually happened?
Processing failed with an error due to EXIF metadata being broken. Looking at exif information, I see it contains a duplicate `Exif\x00\x00Exif\x00\x00` at the beginning. This image is uploaded by one of the clients, so I don't know how exactly that happened. 

`b'Exif\x00\x00Exif\x00\x00II*\x00\x08\x00\x00\x00\x06\x00\x12\x01\x03\x00\x01\x00\x00\x00\x01\x00\x00\x00\x1a\x01\x05\x00\x01\x00\x00\x00V\x00\x00\x00\x1b\x01\x05\x00\x01\x00\x00\x00^\x00\x00\x00(\x01\x03\x00\x01\x00\x00\x00\x02\x00\x00\x00\x13\x02\x03\x00\x01\x00\x00\x00\x01\x00\x00\x00i\x87\x04\x00\x01\x00\x00\x00f\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00\x01\x00\x00\x00H\x00\x00\x00\x01\x00\x00\x00\x06\x00\x00\x90\x07\x00\x04\x00\x00\x000210\x01\x91\x07\x00\x04\x00\x00\x00\x01\x02\x03\x00\x00\xa0\x07\x00\x04\x00\x00\x000100\x01\xa0\x03\x00\x01\x00\x00\x00\xff\xff\x00\x00\x02\xa0\x04\x00\x01\x00\x00\x00\x08\x07\x00\x00\x03\xa0\x04\x00\x01\x00\x00\x00\x08\x07\x00\x00\x00\x00\x00\x00'`

exiftool shows there is error with exif 
`Warning                         : Malformed APP1 EXIF segment`

### What are your OS, Python and Pillow versions?

* OS: MacOS
* Python: 3.10
* Pillow: 10.2.0

```text
Please paste here the output of running:
Pillow 10.2.0
Python 3.10.14 (main, Mar 19 2024, 21:46:16) [Clang 15.0.0 (clang-1500.3.9.4)]
--------------------------------------------------------------------
Python modules loaded from /opt/homebrew/lib/python3.10/site-packages/PIL
Binary modules loaded from /opt/homebrew/lib/python3.10/site-packages/PIL
--------------------------------------------------------------------
--- PIL CORE support ok, compiled for 10.2.0
--- TKINTER support ok, loaded 8.6
--- FREETYPE2 support ok, loaded 2.13.3
--- LITTLECMS2 support ok, loaded 2.16
--- WEBP support ok, loaded 1.4.0
--- WEBP Transparency support ok
--- WEBPMUX support ok
--- WEBP Animation support ok
--- JPEG support ok, compiled for libjpeg-turbo 3.0.2
--- OPENJPEG (JPEG2000) support ok, loaded 2.5.2
--- ZLIB (PNG/ZIP) support ok, loaded 1.2.12
--- LIBTIFF support ok, loaded 4.6.0
*** RAQM (Bidirectional Text) support not installed
*** LIBIMAGEQUANT (Quantization method) support not installed
--- XCB (X protocol) support ok
```

```python
image = Image.open("broken_exif.jpg")
image = ImageOps.exif_transpose(image)
```

Image itself 
broken_exif

### PR DESCRIPTION
**Pull request by zakajd**
Fixes #8348 

Changes proposed in this pull request:

 * Explicitly check for double exif info and drop it

### CODE REVIEW

File: src/PIL/Image.py [hunk 1]
```diff
@@ -3968,7 +3968,9 @@ def load(self, data: bytes) -> None:
         self._data.clear()
         self._hidden_data.clear()
         self._ifds.clear()
-        if data and data.startswith(b"Exif\x00\x00"):
+        if data and data.startswith(b"Exif\x00\x00Exif\x00\x00"):
+            data = data[12:]
+        elif data and data.startswith(b"Exif\x00\x00"):
```
Comment (radarhere) [inline src/PIL/Image.py:3973]:
```suggestion
        while data and data.startswith(b"Exif\x00\x00"):
```
This seems simpler to me.

Comment (zakajd) [reply src/PIL/Image.py]: Done. Also added test for this problem as well

**Comment by radarhere** [comment]
In the issue, you said
> This image is uploaded by one of the clients

Are you sure that this is an image that can be distributed under the Pillow license?

**Comment by zakajd** [comment]
> In the issue, you said
> 
> > This image is uploaded by one of the clients
> 
> Are you sure that this is an image that can be distributed under the Pillow license?

Good point. I'm not a lawyer, to be safe will manually create an image where the EXIF is broken same way.

**Review by radarhere — APPROVED** [review]

**Pull request merged**